noir/noir-repo/compiler/noirc_evaluator/src/brillig/brillig_ir/procedures/array_copy.rs

```diff
@@ -0,0 +1,98 @@
+use std::vec;
+
+use acvm::{acir::brillig::MemoryAddress, AcirField};
+
+use super::ProcedureId;
+use crate::brillig::brillig_ir::{
+    brillig_variable::{BrilligArray, BrilligVariable, BrilligVector, SingleAddrVariable},
+    debug_show::DebugToString,
+    registers::{RegisterAllocator, ScratchSpace},
+    BrilligBinaryOp, BrilligContext,
+};
+
+impl<F: AcirField + DebugToString, Registers: RegisterAllocator> BrilligContext<F, Registers> {
+    /// Conditionally copies a source array/vector to a destination array/vector.
+    /// If the reference count of the source array/vector is 1, then we can directly copy the pointer of the source array/vector to the destination array/vector.
+    pub(crate) fn call_array_copy_procedure(
+        &mut self,
+        source_variable: BrilligVariable,
+        destination_variable: BrilligVariable,
+    ) {
+        // Args
+        let source_pointer = MemoryAddress::from(ScratchSpace::start());
+        let size_register = MemoryAddress::from(ScratchSpace::start() + 1);
+        let is_rc_one_register = MemoryAddress::from(ScratchSpace::start() + 2);
```

Comment:
Seems like we write to this and then don't do anything with it?

Reply:
We do, the procedure branches depending on rc being one (copy vs return the same)
```
brillig_context.codegen_branch(is_rc_one_register, |ctx, cond| {...
```